### PR TITLE
Update OPI Based on IOC Refactor

### PR DESCRIPTION
**Pull request by JackEAllen**
### Description of work

Updated the OPI field values to reflect changes made in [Db refactor](https://github.com/ISISComputingGroup/EPICS_PearlPressure/pull/3) in EPICS_PearlPressure repository so that the OPI displays PV values

### Acceptance criteria

- [ ] PEARLPC IOC OPI displays correct values in appropriate fields for changes made in [Db refactor](https://github.com/ISISComputingGroup/EPICS_PearlPressure/pull/3) as were previously displayed before changes were made to IOC.

### Documentation

* [WIKI](https://github.com/ISISComputingGroup/IBEX/wiki/PEARL-Instrument-Details#pearl-notes)
* IOC PR that changes made in this PR are for: [Db_refactor](https://github.com/ISISComputingGroup/EPICS_PearlPressure/pull/3)
* Testing instructions:
    * Checkout [main](https://github.com/ISISComputingGroup/EPICS_PearlPressure) in `C:\Instrument\Apps\EPICS\support\PearlPressureController\master` and run `make` to ensure they are up to date.
    * Run the tests with `- a` flag to run IOC in simulation mode: `system_tests\run_tests.bat -a`
    * Checkout this PR's branch `opi_refactor` in `C:\Instrument\Dev\ibex_gui`
    * Open `ibex.product` in Eclipse IDE and launch IBEX client after first cleaning project and potentially reloading `targetplatform.target` if needed.
    * From launched IBEX client, add a new device screen for PEARLPC OPI and check OPI behaves as desired by playing around with the controls and current settings, checking that values are populated when set. 
---

#### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi [hunk 1]
```diff
@@ -271,24 +271,24 @@ $(pv_value)</tooltip>
...
           <format_type>0</format_type>
           <height>20</height>
           <horizontal_alignment>0</horizontal_alignment>
           <name>Min_Pressure_Text</name>
           <precision>0</precision>
-          <precision_from_pv>true</precision_from_pv>
-          <pv_name>$(PV_ROOT_PEARLPC):BUFFER2.A</pv_name>
-          <pv_value/>
+          <precision_from_pv>false</precision_from_pv>
+          <pv_name>$(PV_ROOT_PEARLPC):STATUS_ARRAY.[0]</pv_name>
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi:282]:
```suggestion
          <pv_name>$(PV_ROOT_PEARLPC):STATUS_ARRAY.[11]</pv_name>
```

File: base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi [hunk 2]
```diff
@@ -405,24 +405,24 @@ $(pv_value)</tooltip>
...
           </foreground_color>
           <format_type>0</format_type>
           <height>20</height>
           <horizontal_alignment>0</horizontal_alignment>
           <name>Nominal_Pressure_Text</name>
           <precision>0</precision>
           <precision_from_pv>true</precision_from_pv>
-          <pv_name>$(PV_ROOT_PEARLPC):BUFFER2.B</pv_name>
-          <pv_value/>
+          <pv_name>$(PV_ROOT_PEARLPC):STATUS_ARRAY.[1]</pv_name>
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi:416]:
```suggestion
          <pv_name>$(PV_ROOT_PEARLPC):STATUS_ARRAY.[12]</pv_name>
```

File: base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi [hunk 3]
```diff
@@ -653,24 +653,24 @@ $(pv_value)</tooltip>
...
           </foreground_color>
           <format_type>0</format_type>
           <height>20</height>
           <horizontal_alignment>0</horizontal_alignment>
           <name>Max_Pressure_Text</name>
           <precision>0</precision>
           <precision_from_pv>true</precision_from_pv>
-          <pv_name>$(PV_ROOT_PEARLPC):BUFFER2.C</pv_name>
-          <pv_value/>
+          <pv_name>$(PV_ROOT_PEARLPC):STATUS_ARRAY.[2]</pv_name>
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi:664]:
```suggestion
          <pv_name>$(PV_ROOT_PEARLPC):STATUS_ARRAY.[13]</pv_name>
```

File: base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi [hunk 4]
```diff
@@ -3369,24 +3369,24 @@ $(pv_value)</tooltip>
...
           </foreground_color>
           <format_type>0</format_type>
           <height>34</height>
           <horizontal_alignment>0</horizontal_alignment>
           <name>Interface_Status_Text</name>
           <precision>0</precision>
           <precision_from_pv>true</precision_from_pv>
-          <pv_name>$(PV_ROOT_PEARLPC):record_name</pv_name>
-          <pv_value/>
+          <pv_name>$(PV_ROOT_PEARLPC):STATUS_ARRAY</pv_name>
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi:3380]: This is just going to display an array of integers. Doesn't seem very readable, is this really what we want here?

**Review by ThomasLohnert — CHANGES_REQUESTED** [review]

File: base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi (hunk 4 above)
Comment (JackEAllen) [reply base/uk.ac.stfc.isis.ibex.opis/resources/PEARLPC.opi:3380]: I agree about the readability, but I believe that's what's supposed to be returned from looking at the LabVIEW VI panel mentioned [HERE](https://github.com/ISISComputingGroup/IBEX/issues/6040#issuecomment-889145254).

**Comment by JackEAllen** [comment]
Thanks for the review @ThomasLohnert  I've updated the status PV references for min, nominal and max records. Must have forgot to save them before committing first time round. 😅

**Review by ThomasLohnert — APPROVED** [review]

**Pull request merged**